letsencrypt/plugins/webroot.py

```diff
@@ -98,6 +99,10 @@ def prepare(self):  # pylint: disable=missing-docstring
                          self.full_roots[name])
             try:
                 os.makedirs(self.full_roots[name])
+                # Set permissions as parent directory (GH #1389)
+                filemode = stat.S_IMODE(os.stat(path).st_mode)
+                os.chmod(self.full_roots[name], filemode)
```

Comment:
Question: is the reason to call `os.makedirs(file)` then `os.chmod(file, filemode)` rather than just `os.makedirs(file, filemode)` because `makedirs` is affected by umask, while `chmod` is not?  If yes, maybe comment that; if not, just pass filemode to `makedirs`.